x/ecocredit/server/core/features/msg_create_project.feature

```diff
@@ -0,0 +1,31 @@
+Feature: CreateProject
+
+  Background:
+    Given alice has created a credit class
+
+  Rule: a custom project id must be unique
+
+    Scenario: a project id is unique
+      When alice creates a project with id "P1"
+      Then the project exists with id "P1"
+
+    Scenario: a project id is not unique
+      Given alice has created a project with id "P1"
+      When alice creates a project with id "P1"
+      Then expect the error "unique key violation"
+
+  Rule: an auto-generated project id is always unique
+
+    Scenario Outline: a project id is auto-generated based on the project sequence number
+      Given the project sequence number is "<next_sequence>"
+      When alice creates a project
+      Then the project exists with id "<project_id>"
+
+      Examples:
+        | next_sequence | project_id  |
+        | 1             | 001         |
+        | 2             | 002         |
+        | 10            | 010         |
+        | 100           | 100         |
+        | 1000          | 1000        |
+        | 10000         | 10000       |
```

Comment:
cc @aaronc can you confirm this is the intended format for an auto-generated project id?

Reply:
👍

Reply:
This causes an issue with unique project ids when auto-generated. A project can be created with an auto-generated id but a second project within a second credit class (the first project of the second credit class) cannot be created with an auto-generated id because the auto-generated id would be `001`, i.e. the same project id as the first project.

I think we should include the class id in the project id, i.e. `C01-001` and use the project id rather than the class id and project id when formatting the batch denom (the project id includes the class id). See https://github.com/regen-network/regen-ledger/issues/1032#issuecomment-1107800763.

Reply:
This causes a unique constraint violation.

A project can be created under one credit class with an auto-generated id but a second project under another credit class cannot (the first project of the second credit class) because the auto-generated id would be `001`, i.e. the same project id as the first project.

I think we should include the class id in the project id, i.e. `C01-001` and use the project id rather than the class id and project id when formatting the batch denom (the project id includes the class id). See https://github.com/regen-network/regen-ledger/issues/1032#issuecomment-1107800763.

Reply:
This introduces a new issue.

A project can be created under one credit class with an auto-generated id but another project under another credit class (i.e. the first project of the second credit class) cannot be created with an auto-generated id because the auto-generated id for both projects would be `001`.

I think we should include the class id in the project id, i.e. `C01-001` and use the project id rather than the class id and project id when formatting the batch denom (the project id includes the class id). See https://github.com/regen-network/regen-ledger/issues/1032#issuecomment-1107800763.